### DISCUSSION TITLE
passwordless authentication flow

### DISCUSSION THREAD

**Discussion by vesparny** [invertase/react-native-firebase#4903] [Q&A]
Hi, I've implemented the passwordless flow and it works well

What I am doing is, instead of listening to `onAuthStateChange` I simply mark the user as logged in after successful `signInWithEmailLink`

Is store a `loggedIn` boolean in my store and I send the uid token anytime I perform a request to my api, which checks the token validity.
In case it's not valid anymore, I catch the error response and I force a logout in the app.

Is what I am doing ok?

My doubt is I should probably use the `onAuthStateChange` callback to mark the user as logged but my feeling is if `signInWithEmailLink` I am sure the user has a session.

thanks

**Comment by mikehardy** [comment]
This probably best asked on stackoverflow

**Comment by bryanprimus** [comment]
is there an example for this ? or is this even supported by this package?
